### ISSUE TITLE
Road to 0.1.0...

### ISSUE THREAD

**Issue by jasonkriss** [pytorch/ignite#105]
@alykhantejani I wanted to start the discussion of what work is left to do before we can cut an official release (with conda and pypi packages as well). We've used this alpha phase well to work out the fundamental APIs.  But I think its time to give users some guarantees of a stable API so they can have confidence that things aren't gonna change/break underneath them.

Here is a rough sketch of what I feel is blocking us from cutting a 0.1.0 release. It's mostly documentation related.

### Todo
- [ ] convert all docstrings to google style
- [ ] flesh out remaining docstrings
- [ ] update README
  - I'd prefer to take the torchvision approach here and make the README very minimal. Just installation, link to docs, etc...
- [ ] sphinx docs generation
- [ ] sphinx docs hosting
- [ ] release plan
  - where (pypi and conda?)
  - name (looks like ignite is taken on pypi)

What do you think? Do you agree with the overall thoughts? Any disagreements on the todo list or anything missing in your eyes?

**Comment by vfdev-5** [comment]
Release pypi name propositions :
- torchignite
- torchignition <= torchvision :)

[ignite](https://pypi.python.org/pypi/ignite) and [ignition](https://pypi.python.org/pypi/ignition/0.3) are taken

**Comment by vfdev-5** [comment]
Wondering whether current API is stabilized, @jasonkriss could you share your cases of usage of *ignite* if any ?

**Comment by alykhantejani** [comment]
Hi @jasonkriss,

I share the same sentiments about cutting a release. I think the checklist looks fine. I like the idea of a minimal README, however not sure what the best way to explain the concepts of handlers, state etc are.

@soumith can we host the sphinx docs at http://pytorch.org/docs/ similar to torchvision? Also can we put the conda tar under the pytorch channel on conda.

As for pypi, I guess we have to find a new name for that maybe `torchignite`

**Comment by jasonkriss** [comment]
@vfdev-5 @alykhantejani torchignite seems like the obvious choice (following torchtext and torchvision).

@vfdev-5 Sorry I'm not sure I understand what you are asking exactly. 

@alykhantejani What do you think of splitting the docs in to 'Notes' and 'Package Reference' the way pytorch does. Then we can put together a 'Concepts' doc under 'Notes' that will explain all these things. I picture adding a quickstart doc under 'Notes' as well.

@soumith @alykhantejani hosting the docs on pytorch.org/docs would def be ideal IMO. If that's a possibility and there are some things we need to do to conform, I'm happy to take that on.

**Comment by jasonkriss** [comment]
One other option on name is `igniteio`. That's a pretty common pattern when package names are taken and might help us avoid confusion as to whether the package name is `ignite` or actually `torchignite`. Just an idea.

**Comment by vfdev-5** [comment]
@jasonkriss I was wondering if you use *ignite* in some of your use-cases? Such use-cases can reveal whether the API is lack of something. 

I think it could be good to add more various examples on, for example, detection, segmentation tasks etc to show more use-cases in addition to MNIST.

**Comment by DNGros** [comment]
Seems like a good list. I'm excited to see this reach a first release!

`torchignite` seems like a great name choice. To throw out a different idea, something more explicit and boring like `torchtrain` could also make sense with the `torchvision` and `torchtext` scheme. Definitely not as cool sounding though and may be more much contentious...

It might be valuable to add an example of a more complex training scheme such as GAN training or RL training. One of the points made in the README to differentiate from tnt was the higher level of abstraction that can make it easier to train these less traditional models. As @vfdev-5 said, this could potentially reveal if anything in the API is lacking.

Everything seems functional and useful now though, so this definitely something that could wait for a future release.

**Comment by alykhantejani** [comment]
I think writing a few more examples is a good idea, just to check the API is sufficient. Perhaps a regression task (super resolution, segmentation etc.) and GANs would be good use cases to check the API. 

I will also create issues/tasks for:
- converting docstrings to google styles
- stripping down the README
- Creating notes
- Hosting docs (already exists #9)
- creating pypi & conda builds

I guess it's also a good idea to push these examples to the repo (for now at least) to provide users with some examples, until the library is adopted further.

**Comment by alykhantejani** [comment]
Closing this, let's continue task specific discussions on the related issues with label `0.1.0`

**Comment by vfdev-5** [comment]
How about to add code coverage badge too ? For example with [coveralls](https://coveralls.io/).

**Comment by vfdev-5** [comment]
@alykhantejani can we use github "Projects" to follow 0.1.0 roadmap and the todo list from this issue ?

**Comment by jasonkriss** [comment]
@vfdev-5 gotcha. So far i've been using ignite for image classification and detection and some document classification. I think all of these tasks look fairly similar (at the ignite level) to the mnist example. The one thing I have found a lil lacking is coverage of metrics for non-classification tasks. That's something I want to address but it's not a blocker for a release and wont change any apis.

I do agree that more examples would be nice though. I agree with @DNGros and @alykhantejani that  a GAN example would be really helpful as one of the big selling points of ignite is its flexibility to train multiple models.

@alykhantejani Thanks for splitting out those issues!
